From 5fb015f982ed7d76242c0ce1b8132ccc1d42373f Mon Sep 17 00:00:00 2001
From: Michael Stapelberg <stapelberg@debian.org>
Date: Mon, 2 Dec 2013 15:53:34 +0100
Subject: [PATCH] have autocutsel set WM_NAME and friends

---
 autocutsel.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/autocutsel.c b/autocutsel.c
index b04fd25..9342574 100644
--- a/autocutsel.c
+++ b/autocutsel.c
@@ -355,6 +355,35 @@ int main(int argc, char* argv[])
   XtAppAddTimeOut(context, options.pause, timeout, 0);
   XtRealizeWidget(top);
   XUnmapWindow(XtDisplay(top), XtWindow(top));
+
+  /* Set up window properties so that the PID of the relevant autocutsel
+   * process is known and autocutsel windows can be identified as such when
+   * debugging. */
+  const int _NET_WM_NAME = 0;
+  const int UTF8_STRING = 1;
+  const int WM_NAME = 2;
+  const int STRING = 3;
+  const int _NET_WM_PID = 4;
+  const int CARDINAL = 5;
+
+  Atom atoms[6];
+  char *names[6] = {
+      "_NET_WM_NAME",
+      "UTF8_STRING",
+      "WM_NAME",
+      "STRING",
+      "_NET_WM_PID",
+      "CARDINAL",
+  };
+  XInternAtoms(dpy, names, 6, 0, atoms);
+
+  const char *window_name = "autocutsel";
+  pid_t pid = getpid();
+
+  XChangeProperty(dpy, XtWindow(top), atoms[_NET_WM_NAME], atoms[UTF8_STRING], 8, PropModeReplace, (const unsigned char*)window_name, strlen(window_name));
+  XChangeProperty(dpy, XtWindow(top), atoms[WM_NAME], atoms[STRING], 8, PropModeReplace, (const unsigned char*)window_name, strlen(window_name));
+  XChangeProperty(dpy, XtWindow(top), atoms[_NET_WM_PID], atoms[CARDINAL], 32, PropModeReplace, (const unsigned char*)&pid, 1);
+
   XtAppMainLoop(context);
   
   return 0;
-- 
1.8.4.4

