Description: fixes cutsel segfault for empty cut buffer
Author: Jurij Smakov <jurij@wooyd.org>
Bug-Debian: http://bugs.debian.org/727775

diff -aur a/cutsel.c b/cutsel.c
--- a/cutsel.c	2006-11-05 09:25:50.000000000 +0000
+++ b/cutsel.c	2013-10-26 17:34:57.383765419 +0100
@@ -213,7 +213,8 @@
       XtAppAddTimeOut(context, 10, Exit, 0);
     } else {
       options.value = XFetchBuffer(dpy, &options.length, buffer);
-      printf("%s\n", options.value);
+      if (options.length)
+        printf("%s\n", options.value);
       exit(0);
     }
   } else if (strcmp(argv[1], "sel") == 0) {
